src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,198 @@
+                                this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                getAlterationData(this.storeForAllData!.samples.result,
+                                  this.storeForAllData!.patients.result,
+                                  this.storeForAllData!.coverageInformation.result,
+                                  this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                                  this.storeForAllData!.sequencedPatientKeysByGene.result!,
+                                  this.storeForAllData!.selectedMolecularProfiles.result!,
+                                  alterationData,
+                                  false,
+                                  this.alterationFrequencyData,
+                                  this.props.store.genes.result!);
+                
+            });
+
+            this.addGenomicData(this.alterationFrequencyData);
+            // Toasts are removed with delay
+            setTimeout(() => {this.activeToasts.forEach(tId => {toast.dismiss(tId);});}, 1000);
+        }
+
+        return(
+
+            <div className="pathwayMapper">
+            <div data-test="pathwayMapperTabDiv" className="cBioMode" style={{width: "99%"}}>
+                <Row>
+
+                    { !this.isLoading ? 
+
+                        <React.Fragment>
+                            <OqlStatusBanner className="coexp-oql-status-banner" store={this.props.store} tabReflectsOql={true}/>
+
+                            <PathwayMapper
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={this.props.store.genes.result as any}
+                                cBioAlterationData={this.alterationFrequencyData}
+                                queryParameter={QueryParameter.GENE_LIST}
+                                oncoPrintTab={ResultsViewTab.ONCOPRINT}
+                                changePathwayHandler={this.changePathwayHandler}
+                                addGenomicDataHandler={this.addGenomicDataHandler}
+                                tableComponent={PathwayMapperTable}
+                                validGenes={this.validGenes}
+                                toast={toast}
+                            />
+                            <ToastContainer/>
+                        </React.Fragment>
+                    :   <LoadingIndicator isLoading={true} size={"big"} center={true}/>
+                    }
+                </Row>
+            </div>
+            </div>);
+    }
+
+    @computed get storeForAllData(){
+        // Currently Pathways (PathwayMapper) tab must be active, otherwise; all toasts must be closed.
+        if(this.props.store.tabId !== ResultsViewTab.PATHWAY_MAPPER){
```

Comment:
ideally compted functions should not mutate state like this.  they should just return transformations (computations) of their data dependencies.